extensions/amp-story-dev-tools/0.1/amp-story-dev-tools.js

```diff
@@ -32,21 +39,185 @@ const fontsToLoad = [
+
+    this.win.document.title = `Story Dev-Tools (${this.win.document.title})`;
+
+    // TODO: Remove support for url queryParam when widely available.
+    /** @private {string} */
+    this.storyUrl_ = hashParams['url'] || this.win.location.href.split('#')[0];
+
+    /** @private {!DevToolsTab} get URL param for tab (eg: #tab=page-experience) or default to PREVIEW*/
+    this.currentTab_ =
+      Object.values(DevToolsTab).find(
+        (e) => e.toLowerCase().replace(' ', '-') === hashParams['tab']
+      ) || DevToolsTab.PREVIEW;
+
+    /** @private {!Object} maps tabs to contents */
+    this.tabContents_ = {
+      [DevToolsTab.PREVIEW]: new AmpStoryDevToolsTab(
+        createTabContentsTemplate(this.element, DevToolsTab.PREVIEW),
+        this.win,
+        this,
+        this.storyUrl_
+      ),
+      [DevToolsTab.ERRORS]: new AmpStoryDevToolsTab(
+        createTabContentsTemplate(this.element, DevToolsTab.ERRORS),
+        this.win,
+        this,
+        this.storyUrl_
+      ),
+      [DevToolsTab.PAGE_EXPERIENCE]: new AmpStoryDevToolsTab(
+        createTabContentsTemplate(this.element, DevToolsTab.PAGE_EXPERIENCE),
+        this.win,
+        this,
+        this.storyUrl_
+      ),
+    };
   }
 
   /** @override */
   buildCallback() {
     this.loadFonts_();
 
-    // TODO(mszylkowski): Create actual layout.
-    const message = this.element.ownerDocument.createElement('div');
-    message.textContent = 'AMP Story Dev-tools was loaded, page WIP';
-    this.element.textContent = '';
-    this.element.appendChild(message);
+    this.buildLayout_();
+  }
+
+  /**
+   * @private
+   */
+  buildLayout_() {
+    const container = buildContainerTemplate(this.element);
+    this.element.appendChild(container);
+    this.element
+      .querySelector('.i-amphtml-story-dev-tools-close')
+      .addEventListener('click', () => {
+        this.win.location.href = this.storyUrl_;
+      });
+
+    // Create tabs on top
+    const tabsContainer = container.querySelector(
+      '.i-amphtml-story-dev-tools-tabs'
+    );
+    Object.values(DevToolsTab).forEach((e) => {
+      const tab = this.win.document.createElement('span');
+      tab.textContent = e;
+      tab.addEventListener('click', () => {
+        if (this.currentTab_ != e) {
+          this.switchTab_(e);
+          // Update hashString with tab selected or null if tab = preview (default)
+          this.updateHash({
+            'tab':
+              e == DevToolsTab.PREVIEW
+                ? null
+                : e.toLowerCase().replace(' ', '-'),
+          });
+        }
+      });
+      tabsContainer.appendChild(tab);
+    });
+    this.switchTab_(this.currentTab_);
+  }
+
+  /**
+   * Switches the tab shown by activating the tab button and switching the contents.
+   * @param {!DevToolsTab} tab
+   */
+  switchTab_(tab) {
+    this.currentTab_ = tab;
+    toArray(
+      this.element.querySelector('.i-amphtml-story-dev-tools-tabs').children
+    ).forEach((e) => {
```

Comment:
nit: I wonder if you could save a reference to dom nodes generated at line 169, then use that here instead of querying the dom.

Reply:
Good idea, I can actually get that from `this.tabContents_[this.currentTab_]`